doc/src/manual/performance-tips.md

```diff
@@ -1174,31 +1174,31 @@ of a one-dimensional array, and then evaluates the L2-norm of the result:
 
 ```julia
 function init!(u)
-    n = length(u)
-    dx = 1.0 / (n-1)
-    @fastmath @inbounds @simd for i in 1:n
+    inds = linearindices(u)
+    dx = 1.0 / (length(inds)-1)
+    @fastmath @inbounds @simd for i in inds
         u[i] = sin(2pi*dx*i)
     end
 end
 
 function deriv!(u, du)
-    n = length(u)
-    dx = 1.0 / (n-1)
-    @fastmath @inbounds du[1] = (u[2] - u[1]) / dx
-    @fastmath @inbounds @simd for i in 2:n-1
+    inds = linearindices(u)
+    dx = 1.0 / (length(inds)-1)
+    @fastmath @inbounds du[inds[1]] = (u[inds[2]] - u[inds[1]]) / dx
```

Comment:
In most cases the more general style is at least as clear - but in this one case it isn't, and I'm unsure whether this might confuse readers? I tried to be as clear as possible.